providers/gitlab.go

```diff
@@ -213,8 +256,37 @@ func (p *GitLabProvider) EnrichSessionState(ctx context.Context, s *sessions.Ses
 		return fmt.Errorf("group membership check failed: %v", err)
 	}
 
+	p.addProjectMembershipToState(ctx, s)
+
 	s.User = userInfo.Username
 	s.Email = userInfo.Email
 
 	return nil
+
+}
+
+// addProjectMembership adds projects into session.Groups
+func (p *GitLabProvider) addProjectMembershipToState(ctx context.Context, s *sessions.SessionState) {
+	if len(p.Projects) == 0 {
+		return
+	}
+
+	// Iterate over projects, check if oauth2-proxy can get project information on behalf of the user
+	for _, project := range p.Projects {
+		project, err := p.getProjectInfo(ctx, s, project)
+		// If no error, it's a member of the project
+		if err == nil && !project.Archived {
+			s.Groups = append(s.Groups, project.PathWithNamespace)
+		}
+
+		if err != nil {
+			logger.Printf("Warning: project info request failed: %v", err)
+		}
+
+		if project != nil && project.Archived {
+			logger.Printf("Warning: project %s is archived", project.Name)
```

Comment:
This might be noisy? This will log for every user login on an archived project, right?

I think in both these cases if we stick with it you want `Errorf` too?

Reply:
How errors are logged is other providers ? I think it's important to know why user is not authorized to see the ressource.

Reply:
Good point, it seems for the most part in similar situations in Google & Github where i see a similar pattern, they log.